### PR TITLE
feat: add interactive script for new peerings

### PR DESCRIPTION
**Pull request by yzguy**
Addresses #59

Interactive script for new peerings

```
-> ./interactive.py

What router would you like to peer at?

	1. router.ams1  	8. router.fra1   	15. router.mia1  	22. router.sto1
	2. router.atl1  	9. router.gru1   	16. router.mil1  	23. router.syd1
	3. router.cgk1  	10. router.iad1  	17. router.mum1  	24. router.tor1
	4. router.chi1  	11. router.lax1  	18. router.osa1  	25. router.tyo1
	5. router.dal1  	12. router.lon1  	19. router.par1
	6. router.ewr1  	13. router.maa1  	20. router.sea1
	7. router.fnc1  	14. router.mad1  	21. router.sin1

Selection: 7

Peer Name: YZGUY-FNC

DN42 ASN: 4242421669

Which BGP peering type are you requesting?

	1. Multi-protocol (IPv4 and IPv6) with Extended Nexthop capability (preferred)
	2. Multi-protocol (IPv4 and IPv6)
	3. IPv4 and IPv6 (separate sessions)
	4. IPv4 ONLY
	5. IPv6 ONLY

Selection: 1

IPv6 tunnel address (Link-local preferred, /64 assumed unless specified): fe80::1669

Wireguard Endpoint: rtr1.mkc1.as199344.net

Wireguard Port: 20207

Wireguard Public Key: 2BEC3CFKpAiIK3wTStbVuHyy894p5g2JLLTajdBsUF8=

Validating peer: YZGUY-FNC... ok 

Successfully saved peer to router.fnc1.yml
```

Alternatively, if you run `./interactive.py --stdout` it will output to STDOUT instead of write to file

```
--- Generated Configuration ---

# routers/router.fnc1.yml

- name: YZGUY-FNC
  asn: 4242421669
  ipv6: fe80::1669
  multiprotocol: true
  extended_nexthop: true
  sessions:
    - ipv6
  wireguard:
    remote_address: rtr1.mkc1.as199344.net
    remote_port: 20207
    public_key: 2BEC3CFKpAiIK3wTStbVuHyy894p5g2JLLTajdBsUF8=
```

### CODE REVIEW

**Comment by yzguy** [comment]
Load the file, sort by `name` key, then write back out the file with `yaml.dump(.., sort_keys=False)`

**Comment by yzguy** [comment]
PyYAML will not preserve comments, we would need to use `ruamel.yaml`

```
routers/router.dal1.yml
13:- name: JK-NETWORK-DAL  # PeerID: 23206

routers/router.tyo1.yml
13:- name: JK-NETWORK-TYO  # PeerID: 23709
```

**Comment by jvoss** [comment]
> PyYAML will not preserve comments, we would need to use `ruamel.yaml`
> 
> 
> 
> ```
> 
> routers/router.dal1.yml
> 
> 13:- name: JK-NETWORK-DAL  # PeerID: 23206
> 
> 
> 
> routers/router.tyo1.yml
> 
> 13:- name: JK-NETWORK-TYO  # PeerID: 23709
> 
> ```

These comments are not critical and can go away if necessary. If we do have a reason to document something it should probably exist elsewhere anyways.

**Review by jvoss — COMMENTED** [review]
Is there a way to use multiple columns for questions that have a lot of options? Running this on a standard window size (80x24) requires some scrolling. 

Maybe something like?
```
What router would you like to peer at?

	1. router.ams1		10. router.iad1		19. router.par1
	2. router.atl1		11. router.lax3		20. router.sea1
	3. router.cgk1		12. router.lon1		21. router.sin1
	4. router.chi1		13. router.maa1		22. router.sto1
	5. router.dal1		14. router.mad1		23. router.syd1
	6. router.ewr1		15. router.mia1		24. router.tor1
	7. router.fnc1		16. router.mil1		25. router.tyo1
	8. router.fra1		17. router.mum1
	9. router.gru1		18. router.osa1
	

Selection:
```

File: interactive.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+    with open(f'routers/{router}.yml', 'w') as fd:
+        # YAML document header
+        fd.write('---\n')
+        # Write out each peer to router file (do not sort each peers keys)
+        for peer in peers:
+            fd.write(yaml.dump([peer], Dumper=IndentDumper, sort_keys=False))
+            # Don't write newline after last peer
+            if peer != peers[-1]:
+                fd.write('\n')
```
[the remark this answers is not in the document]
Comment (jvoss) [reply interactive.py:65]: Never mind. There is a newline at the end of the file but not a complete empty new-line and I'm cool with this.

File: interactive.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (jvoss) [reply interactive.py:1]: It did work in my own venv without modification but I'm trying to think through edge cases on how users with different environments would use this.

**Comment by yzguy** [comment]
All concerns addressed other than the multi-column output will need to think about how to do that one.

**Comment by yzguy** [comment]
Just need to implement a uniqueness check before we write it to the file

**Review by jvoss — APPROVED** [review]

**Review by jvoss — APPROVED** [review]

**Pull request merged**